apricot/local/service.go

```diff
@@ -122,6 +122,23 @@ func (s *Service) GetDefaults() map[string]string {
 	return smap
 }
 
+func (s *Service) GetHostInventory(keyPrefix string) (hosts []string, err error) {
```

Comment:
Please see my comment further below on `keyPrefix` vs. `detector` as parameter.

I think `keyPrefix` should be a constant `inventoryKeyPrefix` defined at the top of this file, and then if you receive a non-empty `detector`, you amend the prefix you pass to `GetKeysByPrefix` as needed.